### PR TITLE
Fix permissions and add software

### PR DESCRIPTION
**Pull request by tmetzl**
Added a fix for a permission error during installation and added two new packages needed for the AMR course (pyside, gnuplot)

### CODE REVIEW

File: content/ros/ROS Installation/ros_catkin_installer.sh [hunk 1]
```diff
@@ -77,6 +83,7 @@ function create_catkin_workspace {
 	catkin_make
 	echo "Sourcing the workspace"
 	source devel/setup.bash
+	sudo chown -R $SUDO_USER:$SUDO_USER ~/catkin_ws/
```
Comment (minhnh) [inline content/ros/ROS Installation/ros_catkin_installer.sh]: does the workspace in `$HOME` really need to be `chown` to `$SUDO_USER`? Seems excessive.

File: content/ros/ROS Installation/ros_catkin_installer.sh [hunk 2]
```diff
@@ -77,6 +83,7 @@ function create_catkin_workspace {
 	catkin_make
```
Comment (minhnh) [inline content/ros/ROS Installation/ros_catkin_installer.sh]: if possible I suggest migrating to [`catkin_tools`](https://catkin-tools.readthedocs.io/en/latest/), since it's in general much nicer than the old `catkin_make` interface.

File: content/ros/ROS Installation/ros_catkin_installer.sh [hunk 3]
```diff
@@ -21,7 +21,11 @@ function ros_install {
 	echo "Installing Map Server"
 	sudo apt-get -y install ros-kinetic-map-server
 	sudo apt-get -y install python-rosinstall python-rosinstall-generator python-wstool build-essential
+	sudo apt-get -y install python-pyside
+	sudo apt-get -y install gnuplot
 	source /opt/ros/kinetic/setup.bash
+	sudo chown -R $SUDO_USER:$SUDO_USER ~/.ros/
```
Comment (argenos) [inline content/ros/ROS Installation/ros_catkin_installer.sh]: Why is this needed?

File: content/ros/ROS Installation/ros_catkin_installer.sh (hunk 2 above)
Comment (argenos) [reply content/ros/ROS Installation/ros_catkin_installer.sh]: I agree, please change this to catkin tools, you’ll need to add this to the packages that need to be installed (python-catkin-tools)

File: content/ros/ROS Installation/ros_catkin_installer.sh (hunk 1 above)
Comment (argenos) [reply content/ros/ROS Installation/ros_catkin_installer.sh]: Same as above, why is this needed?

**Comment by tmetzl** [comment]
I reorganized the script to make it easier to change the packages installed.
When switching from desktop-full to desktop I had to add Stage.
I switched from catkin_make to catkin tools and added a check at each installation step to make sure the packages are installed properly.
I got rid of the sudo chown because I could not reproduce the error that I had without it anymore.

I might add a few more prompts and checks to make sure the catkin workspace is only generated if the folder catkin_ws is not yet present.

I tested everything on a fresh Ubuntu 16.04 with updates yesterday.

**Comment by tmetzl** [comment]
I had to reintroduce the chown commands.
The problem is that the script needs to be executed as sudo to not ask for a password repeatedly during the installation process. Folders that are created by a script running as sudo are owned by root and can't be modified by the user. This is why the permissions need to be set to the user calling the script for the .ros and catkin_ws folders.

**Comment by minhnh** [comment]
I think it's actually better to execute the script as non-sudo and force the users to enter their passwords everytime they're needed. Though from my experience Ubuntu will only ask for password once. I tested this out with the following script:
```
#!/usr/bin/env sh
sudo apt update
sudo apt upgrade
```

**Comment by tmetzl** [comment]
When you use sudo it will be remembered for a certain amount of time, usually 15 mins. Installing ROS often takes quite a bit longer, especially when your hardware or connection is not that fast.
So after 15 mins it would ask at a random point for the password again without it being clear to the user why it is necessary exactly at this point. Remember that the script is targeted at new students who might not have a lot of Linux experience.

When you force the script to be executed as sudo you only have to give your password once, can select "install everything" and then go for a coffee and come back to a fully installed ROS. Additionally it is not uncommon for installation scripts to require sudo. So in my opinion it is more convenient to do it the way I did.

If you start asking for the password or confirmations more often you can just give the students a list of software and tell them to install it on their own. My intention was to have some sort of one-click installer for less experienced students that takes care of everything so problems during the installation are kept to a minimum.

File: content/ros/ROS Installation/ros_catkin_installer.sh [hunk 4]
```diff
@@ -1,159 +1,284 @@
...
+    else
+      echo ""
+      echo "ERROR: Package $1 could not be installed!"
+      echo "Installation aborted!"
+      exit 1
+    fi
+  fi
+}
 
 function update_keys {
```
Comment (argenos) [inline content/ros/ROS Installation/ros_catkin_installer.sh]: If we aim to reduce the number of times sudo is required, in principle `update_keys` and `ros_install` can be a single function.

**Comment by minhnh** [comment]
resolved with PR #46, this can be closed now with issue #41
